packages/services/src/kernel/default.ts

```diff
@@ -1375,26 +1354,21 @@ export class KernelConnection implements Kernel.IKernelConnection {
           // Updating the status is synchronous, and we call no async user code
           const executionState = (msg as KernelMessage.IStatusMsg).content
             .execution_state;
-          if (executionState === 'autorestarting') {
+          if (executionState === 'restarting') {
             // The kernel has been auto-restarted by the server. After
             // processing for this message is completely done, we want to
             // handle this restart, so we don't await, but instead schedule
             // the work as a microtask (i.e., in a promise resolution). We
             // schedule this here so that it comes before any microtasks that
             // might be scheduled in the status signal emission below.
             void Promise.resolve().then(async () => {
-              // handleRestart changes the status to 'restarting', so we call it
-              // first so that the status won't flip back and forth between
-              // 'restarting' and 'autorestarting'.
-              await this._handleRestart();
               this._updateStatus('autorestarting');
-            });
-          }
-          if (executionState === 'restarting') {
-            void Promise.resolve().then(async () => {
-              await this._handleRestart();
-              this._kernelSession = msg.header.session;
-              this._updateStatus('restarting');
+              this._clearKernelState();
+
+              // We must reconnect since the kernel connection information may have
+              // changed, and the server only refreshes its zmq connection when a new
+              // websocket is opened.
+              await this.reconnect();
```

Comment:
We don't need to request kernel info (as was done in the _handleRestart) because the reconnect will automatically do that. We do have to reconnect any time there was a restart on the server, since that triggers the server to refresh its zmq connection, and for consistency with manual restarts.